azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/presentation/fragment/setup/components/JoinCallButtonHolderView.kt

```diff
@@ -55,10 +56,15 @@ internal class JoinCallButtonHolderView : ConstraintLayout {
         joiningCallText.text = context.getString(R.string.azure_communication_ui_calling_setup_view_button_connecting_call)
 
         setupJoinCallButton.setOnClickListener {
-            if (networkManager.isNetworkConnectionAvailable()) {
-                viewModel.launchCallScreen()
-            } else {
+            val networkAvailable = networkManager.isNetworkConnectionAvailable()
```

Comment:
I see this was followed before this PR, but this data should probably be in the ViewModel, and manager shouldn't be a dependency of the component.

Reply:
yup. Moving this to ViewModel. Thanks.